contracts/SlashIndicator.sol

```diff
@@ -39,22 +51,40 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
     bool exist;
   }
 
+  // Proof that a validator misbehaved in fast finality
+  struct FinalityEvidence {
+    uint256 numA;
+    bytes32 headerA;
+    bytes   sigA;
+    uint256 numB;
+    bytes32 headerB;
+    bytes   sigB;
+    address valAddr;
+  }
+
   modifier oncePerBlock() {
     require(block.number > previousHeight, "can not slash twice in one block");
     _;
     previousHeight = block.number;
   }
 
   modifier onlyZeroGasPrice() {
-    
     require(tx.gasprice == 0 , "gasprice is not zero");
-    
     _;
   }
   
+  modifier noReentrant() {
```

Comment:
Let us do not introduce `noReentrant` here, let's use `onlyRelayer` here to prevent Reentrant Attack, only registered relayer can do submit slash evidence.

Reply:
Fixed